crates/felt-macro/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+
+                match LambdaFieldElement::from_bytes_be(&buffer) {
+                    Ok(field_element) => Ok(HandleExprOutput::ComptimeFelt(field_element)),
+                    Err(_) => Err(Error::new_spanned(
+                        lit_byte_str,
+                        "Failed to convert byte string to Felt",
+                    )),
+                }
+            }
+
+            Lit::Char(lit_char) => {
+                let char = lit_char.value();
+
+                if !char.is_ascii() {
+                    return Err(Error::new_spanned(
+                        lit_char,
+                        "Only ASCII characters are supported",
+                    ));
+                }
+
+                let mut buffer = [0u8];
+                char.encode_utf8(&mut buffer);
+
+                Ok(HandleExprOutput::ComptimeFelt(LambdaFieldElement::from(
+                    &UnsignedInteger::from(u16::from(buffer[0])),
+                )))
+            }
+
+            Lit::Byte(lit_byte) => {
+                let char = lit_byte.value();
+
+                Ok(HandleExprOutput::ComptimeFelt(LambdaFieldElement::from(
+                    &UnsignedInteger::from(u16::from(char)),
+                )))
+            }
+
+            Lit::CStr(_) | Lit::Float(_) | Lit::Verbatim(_) => {
+                Err(Error::new_spanned(expr_lit, "Unsupported literal type"))
+            }
+
+            // `Lit` is a non-exhaustive enum
+            _ => Err(Error::new_spanned(expr_lit, "Unknown literal type")),
+        },
+
+        Expr::Unary(expr_unary) => handle_expr_unary(expr_unary),
+
+        _ => Ok(HandleExprOutput::Runtime),
+    }
+}
+
+fn handle_lit_int(lit_int: &LitInt) -> Result<HandleExprOutput> {
+    let value = match lit_int.base10_parse::<u128>() {
+        Ok(v) => v,
+        Err(_) => {
+            return Err(Error::new_spanned(
+                lit_int,
+                "Invalid integer literal for Felt conversion",
+            ));
+        }
+    };
+
+    Ok(HandleExprOutput::ComptimeFelt(LambdaFieldElement::from(
+        &UnsignedInteger::from(value),
+    )))
+}
+
+fn handle_expr_unary(expr_unary: &ExprUnary) -> Result<HandleExprOutput> {
+    let ExprUnary {
+        attrs: _attrs,
+        op,
+        expr,
+    } = expr_unary;
+    match op {
+        syn::UnOp::Not(_) => match &**expr {
+            Expr::Lit(ExprLit {
+                lit: Lit::Bool(lit_bool),
+                ..
+            }) => Ok(HandleExprOutput::ComptimeFelt(match lit_bool.value() {
+                false => LambdaFieldElement::from(&UnsignedInteger::from_u64(1)),
+                true => LambdaFieldElement::from(&UnsignedInteger::from_u64(0)),
+            })),
+            Expr::Lit(ExprLit {
+                lit: Lit::Int(_lit_int),
+                ..
+            }) => Err(Error::new_spanned(
+                expr,
+                "The `!` logical inversion operator is applicable to the `Felt` type",
```

Comment:
Why the error message in this case is different than the one from line 209?
Why does this macro only accept the `!` operator only with integer variables but doesn't with literals?

Reply:
It's a bit more specific, because `!42` is a valid rust expression, but is invalid in the context of `Felt`, and the `Not` operator doesn't exist in this context.
The second message is about stuff like `!"abc"` which are not even valid Rust expression.

Reply:
to be honest `!b'c'` is a valid rust expression tho. COz it returns a u8.
So this one could be handled in the first one too